docs/quickstart.rst

```diff
@@ -2,335 +2,227 @@
-        }
-    
-        // Delete resources
-        uxr_delete_session(&session);
-        uxr_close_udp_transport(&transport);
-    
-        return 0;
-    }
-
-At this moment, the subscriber will receive the topics that are sending by the publisher.
-
-In order to see the messages from the DDS Global Data Space point of view, you can use *eProsima Fast RTPS* HelloWorld example running a subscriber
-(`Fast RTPS HelloWorld <http://eprosima-fast-rtps.readthedocs.io/en/latest/introduction.html#building-your-first-application>`_): ::
-
-    $ cd /usr/local/examples/C++/HelloWorldExample
-    $ sudo make && cd bin
-    $ ./HelloWorldExample subscriber
 
-Learn More
-----------
 
-To learn more about DDS and *eProsima Fast RTPS*: `eProsima Fast RTPS <http://eprosima-fast-rtps.readthedocs.io>`_
+*eProsima Micro XRCE-DDS* provides a C API which allows the creation of *eProsima Micro XRCE-DDS Clients* that can either
+publish/subscribe to topics from the DDS Global Data Space, or act as client-service applications following
+a request/reply pattern.
+
+In this section, we will guide the user through the deployment of two out-of-the-box examples.
+In the first one, an *eProsima Micro XRCE-DDS Agent*
+bridges two *eProsima Micro XRCE-DDS Clients* publishing and subscribing to the DDS world.
+The second example shows instead the deployment of a client-service application using the
+Requester and Replier entities, also put into communication via an *eProsima Micro XRCE-DDS Agent*.
+
+The section is organized as follows:
+
+- :ref:`run_agent`
+- :ref:`run_pubsub_example`
+- :ref:`run_reqrep_example`
+
+.. _run_agent:
+
+Running an Agent
+----------------
+
+First of all, install the *Agent* as explained in the :ref:`install_agent` section.
+On Linux, this would be: ::
+
+    $ git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
+    $ cd Micro-XRCE-DDS-Agent
+    $ mkdir build && cd build
+    $ cmake ..
+    $ make
+    $ sudo make install
+
+After having installed the *Agent* system-wide, you can launch it.
+
+For this example, the *Client*-*Agent* communication will be done through UDP, using the port :code:`2019`
+and with the XML creation mode, which is the default mode for creating entities: ::
+
+    $ cd /usr/local/bin && MicroXRCEAgent udp -p 2019
+
+.. TODO: write a comment tagging guys and asking if it was ok to remove the "-r <references-file>" part
+
+.. _run_pubsub_example:
+
+Running a Publisher-Subscriber example
+--------------------------------------
+
+In this section, we guide the user through the configuration and deployment of
+a simple publish/subscribe example where the communication is mediated by the *Agent* created above.
+
+Before considering the publisher and subscriber examples, it is useful to briefly summarize how the
+`Publisher` and `Subscriber` entities work, as well as to list the functions related to both entities.
+
+.. TODO: change text below according to the opportune entities.
```

Comment:
@pablogs9

Reply:
Addressed in e01cb842b09074e8409a97657409bcc63c475270